### PR TITLE
add a combine_attrs option to open_mfdataset

### PR DESCRIPTION
**Pull request by keewis**
In order to fix the failing tests in #4902 we need to expose `combine_attrs` to be able to properly construct the expected result (to be passed through to the combine function).

This overlaps with the fallback of the `attrs_file` code, which I removed for now. Maybe `combine_attrs="override"` would be better?

- [x] Tests added
- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by mathause** [comment]
The motivation for `attrs_file` is given in  #2382. Maybe require `combine_attrs == "override"` when `attrs_file is not None`?

The current behavior is `combine_attrs="override"`, right? So when choosing something else this would have to be deprecated

**Comment by keewis** [comment]
for the main object, yes, but for the variables of that object `"drop"` is currently always used. That means that if we try to sync both we introduce a breaking change.

We might eventually want to switch both to `"drop_conflicts"` so we would need to go through a deprecation cycle anyways.

**Comment by keewis** [comment]
(the RTD build fails because `docs.scipy.org` is down)

**Review by mathause — APPROVED** [review]

**Comment by keewis** [comment]
should we merge this? It is blocking #4902, which in turn is blocking #5041.

File: xarray/tests/test_backends.py [hunk 1]
```diff
@@ -3004,6 +3004,60 @@ def test_open_mfdataset_does_same_as_concat(self, combine, opt, join):
...
 
+    @pytest.mark.parametrize(
+        ["combine_attrs", "attrs", "expected", "expect_error"],
+        (
+            pytest.param("drop", [{"a": 1}, {"a": 2}], {}, False, id="drop"),
+            pytest.param(
+                "override", [{"a": 1}, {"a": 2}], {"a": 1}, False, id="override"
+            ),
+            pytest.param(
+                "no_conflicts", [{"a": 1}, {"a": 2}], None, True, id="no_conflicts"
```
Comment (dcherian) [inline xarray/tests/test_backends.py:3015]: Hmm... `expected=None` looks like a bug to me. Shouldn't `.attrs` always return a dictionary?

**Review by dcherian — APPROVED** [review]

File: xarray/tests/test_backends.py (hunk 1 above)
Comment (keewis) [reply xarray/tests/test_backends.py:3015]: this set of parameters has `expect_error=True`, so `expected` is not used at all

Comment (dcherian) [reply xarray/tests/test_backends.py:3015]: Haha. Missed that. Thanks!

**Pull request merged**